### PR TITLE
[6.x] Bumped monolog 2.0

### PR DESCRIPTION
**Pull request by Vendin**
Changelog
https://github.com/Seldaek/monolog/blob/2.0.0/CHANGELOG.md#200-2019-08-30
Upgrage
https://github.com/Seldaek/monolog/blob/2.0.0/UPGRADE.md

### CODE REVIEW

**Comment by X-Coder264** [comment]
As mentioned in https://github.com/laravel/framework/pull/29813 the `illuminate/log` composer.json should be updated too.

We should also probably make the constraint `^1.12|^2.0` to allow for an easier upgrade path to Laravel 6 for the community packages.

**Comment by Vendin** [comment]
@X-Coder264, done, thanks for the comments
https://github.com/illuminate/log/pull/3

**Comment by X-Coder264** [comment]
Np. btw, the `illuminate/log` constraint should be updated through this PR. All PRs to the framework are done through this repository, not through the individual component repository.

**Comment by taylorotwell** [comment]
So we have literally nothing incompatible with Monolog 2.0 in our current code?

**Comment by GrahamCampbell** [comment]
Hmm, will be a pain for people to have to have their code work with both Monolog 1 and 2. I'm thinking in particular for package developers.

**Comment by X-Coder264** [comment]
Why would package developers have to maintain support for both Monolog 1 and 2?

**Comment by GrahamCampbell** [comment]
> Why would package developers have to maintain support for both Monolog 1 and 2?

To say it can be guaranteed to be able to be installed in Laravel 6.

**Comment by taylorotwell** [comment]
Yeah, I think we would want to choose one or the other for this release.

**Comment by GrahamCampbell** [comment]
Maybe it makes sense to wait till Laravel 7 for this, and go straight to v2. Like, v1 works fine, and has loads of packages that already work with it. I don't see any urgency to upgrade from a stable component.

**Comment by Vendin** [comment]
@taylorotwell, file name format must now contain {date} for daily channel, what is the best way to implement this?

```
RotatingFileHandler
The file name format must now contain {date} and the date format must be set to one of the predefined FILE_PER_* constants to avoid issues with file rotation. See setFilenameFormat.
```

```
protected function createDailyDriver(array $config)
    {
        return new Monolog($this->parseChannel($config), [
            $this->prepareHandler(new RotatingFileHandler(
                $config['path'], $config['days'] ?? 7, $this->level($config),
                $config['bubble'] ?? true, $config['permission'] ?? null, $config['locking'] ?? false
            ), $config),
        ]);
    }
```

```
        'daily' => [
            'driver' => 'daily',
            'path' => storage_path('logs/laravel.log'),
            'level' => 'debug',
            'days' => 14,
        ],
```

Alternatively, we can make all the necessary parameters in the configs, and form the file name somewhere in the LogManager, it can be directly in createDailyDriver
```
        'daily' => [
            'driver' => 'daily',
            'directory' => storage_path('logs/'),
            'filename_format' => '{filename}-{date}',
            'filename' => 'laravel'
            'date_format' => 'Y-m-d',
            'level' => 'debug',
            'days' => 14,
        ],
```
Or another option, something like (not very)
```
        'daily' => [
            'driver' => 'daily',
            'path' => storage_path(sprintf('logs/laravel-%s.log', now()->toDateString())),
            'level' => 'debug',
            'days' => 14,
        ],
```

Or generally something different.

You know the best.

File: composer.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
         "egulias/email-validator": "^2.0",
         "erusev/parsedown": "^1.7",
         "league/flysystem": "^1.0.8",
-        "monolog/monolog": "^1.12",
+        "monolog/monolog": "^1.12 || ^2.0",
```
Comment (driesvints) [inline composer.json]: Please use `^1.12|^2.0` notation for constraints.

```suggestion
        "monolog/monolog": "^1.12|^2.0",
```

File: src/Illuminate/Log/composer.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
         "php": "^7.2",
         "illuminate/contracts": "^6.0",
         "illuminate/support": "^6.0",
-        "monolog/monolog": "^1.11"
+        "monolog/monolog": "^1.12 || ^2.0"
```
Comment (driesvints) [inline src/Illuminate/Log/composer.json]: Please use `^1.12|^2.0` notation for constraints.

```suggestion
        "monolog/monolog": "^1.12|^2.0"
```

**Review by driesvints — CHANGES_REQUESTED** [review]

**Comment by driesvints** [comment]
Tests are passing so I don't think there's anything broken? This build uses 2.0: https://travis-ci.org/laravel/framework/jobs/579491706

My vote is to allow both in Laravel 6 and remove support for 1.0 in Laravel 7

**Comment by X-Coder264** [comment]
I agree with @driesvints. I don't see why Laravel 6 couldn't/shouldn't allow both Monolog versions as there doesn't seem to be any additional maintenance burden for the framework for that. It will allow package developers to support whatever Monolog version they want now and it will give them 6 months until Laravel 7 to provide Monolog 2 support if they do not provide it now. Symfony 5 will also support both Monolog versions. Laravel also supported two different major version of Carbon untill recently and there was zero problems with that as far as I know.

File: src/Illuminate/Log/composer.json (hunk 2 above)
Comment (Vendin) [reply src/Illuminate/Log/composer.json]: done

File: composer.json (hunk 1 above)
Comment (Vendin) [reply composer.json]: done

**Comment by gmsantos** [comment]
@Vendin the default filename format for rotating handler already includes the `{date}` configuration.

image

https://github.com/Seldaek/monolog/blob/master/src/Monolog/Handler/RotatingFileHandler.php#L52
https://github.com/Seldaek/monolog/blob/master/src/Monolog/Handler/RotatingFileHandler.php#L84-L96

**Comment by Vendin** [comment]
@gmsantos thanks, forgot about it

> @Vendin the default filename format for rotating handler already includes the `{date}` configuration.
> 
> image
> 
> https://github.com/Seldaek/monolog/blob/master/src/Monolog/Handler/RotatingFileHandler.php#L52
> https://github.com/Seldaek/monolog/blob/master/src/Monolog/Handler/RotatingFileHandler.php#L84-L96

**Review by driesvints — APPROVED** [review]

**Pull request merged**